### PR TITLE
INTMDB-233: Update linter version

### PR DESCRIPTION
**Pull request by MihaiBojin**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
- updated the linter version
- and fixed the config and necessary code

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ linters-settings:
       - octalLiteral
   govet:
     check-shadowing: true
-  golint:
+  revive:
     min-confidence: 0
```
Comment (gssbzn) [inline .golangci.yml]: `min-confidence` is not a valid revive setting, https://github.com/mgechev/revive#configuration

Feel free to check the cli but we have some extra rules that don't come enabled by default 

https://github.com/mongodb/mongocli/blob/6ac1f831b2a7d4ae60f4a7d025e4ab3c8da50d79/.golangci.yml#L22-L53

File: mongodbatlas/data_source_mongodbatlas_third_party_integration_test.go [hunk 2]
```diff
@@ -243,7 +239,8 @@ func testAccCreateThirdPartyIntegrationConfig() *matlas.ThirdPartyIntegration {
 func testGenString(length int, charSet string) string {
 	sequence := make([]byte, length)
 	for i := range sequence {
-		sequence[i] = charSet[seeder.Intn(len(charSet))]
+		n, _ := rand.Int(rand.Reader, big.NewInt(int64(len(charSet))))
```
Comment (gssbzn) [inline mongodbatlas/data_source_mongodbatlas_third_party_integration_test.go]: [nit] can we move `big.NewInt(int64(len(charSet)))` outside the loop given is the same for every iteration?

**Review by gssbzn — COMMENTED** [review]
thanks for doing this, wow this was really behind, I left a couple of nits that may be worth looking into before merging

File: .golangci.yml (hunk 1 above)
Comment (MihaiBojin) [reply .golangci.yml]: updated.

File: mongodbatlas/data_source_mongodbatlas_third_party_integration_test.go (hunk 2 above)
Comment (MihaiBojin) [reply mongodbatlas/data_source_mongodbatlas_third_party_integration_test.go]: I imagined the compiler would be smart enough to do that automatically (plus this is test code), but I made the change anyway.

File: GNUmakefile [hunk 3]
```diff
@@ -11,9 +11,10 @@ GITTAG=$(shell git describe --always --tags)
 VERSION=$(GITTAG:v%=%)
 LINKER_FLAGS=-s -w -X 'github.com/mongodb/terraform-provider-mongodbatlas/version.ProviderVersion=${VERSION}'
 
-GOLANGCI_VERSION=v1.29.0
+GOLANGCI_VERSION=v1.41.1
 
-export PATH := ./bin:$(PATH)
+export PATH := $(shell go env GOPATH)/bin:$(PATH)
+export SHELL := env PATH=$(PATH) /bin/bash
```
Comment (MihaiBojin) [inline GNUmakefile:17]: This line is required to use the PATH on MacOS

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by themantissa — APPROVED** [review]
LGTM!

**Comment by abner-dou** [comment]
LGTM!

**Pull request merged**